Changed layer number and learning rate init to make execution of plot_mnist_filters.py quicker

sveneschlbeck opened the pull request:
#21598 @adrinjalali

Tuned down the number of layers and adapted the ``learning_rate_init`` and ``max_iter`` params to make it between 15 and 20 sec quicker with almost no quality loss and without changing the learning effect or character of the example.
![mnist_before](https://user-images.githubusercontent.com/66701689/141517922-54b9288b-f9db-485d-a929-31d3a2318762.png)
![mnist_after](https://user-images.githubusercontent.com/66701689/141517932-b6640644-b0d4-453c-8ff7-4428d0c12b23.png)

Commit:
Changed layer number and learning rate init to make execution of example faster

ogrisel reviewed (approved):
LGTM, thanks for the improvement.

ogrisel commented:
While you are at it, could you please edit the comments to expand the CI acronym from "because of CI’s time constraints" to "because of resource usage constraints on our Continuous Integration infrastructure that is used to build this documentation on a regular basis."

adrinjalali commented:
The comment in the code says that this doesn't converge, does it still not converge?

There's also a mistake in the comment, there's an extra "are" there, you can fix that too.

Also, please mention the absolute times before and after.

Commit:
Update plot_mnist_filters.py

sveneschlbeck commented:
@ogrisel Edited your improved explanation into the docstring

sveneschlbeck commented:
@adrinjalali Yes, rest should be good, will delete the additional "are" and also paste in some new run times

Commit:
Update plot_mnist_filters.py

sveneschlbeck commented:
@adrinjalali @ogrisel I deleted the additional "are" and added @ogrisel`s comment also in that line since it was the same as in the docstring, so I figured to put the new explanation in both lines (docstring and single line comment)

Commit:
Update plot_mnist_filters.py

adrinjalali commented:
I think we can get a better speedup here if we use a much smaller part of the data as training set. Could you please use `train_test_split`, and use a small part of the data as training, while somewhat keeping the output the same?

sveneschlbeck commented:
@adrinjalali Okay, I'll get on it, soon :)

sveneschlbeck commented:
@adrinjalali We get the following results:

With this code

`import matplotlib.pyplot as plt
from sklearn.datasets import fetch_openml
from sklearn.exceptions import ConvergenceWarning
from sklearn.neural_network import MLPClassifier

# Load data from https://www.openml.org/d/554
X, y = fetch_openml("mnist_784", version=1, return_X_y=True)
X = X / 255.0

# rescale the data, use the traditional train/test split
X_train, X_test = X[:60000], X[60000:]
y_train, y_test = y[:60000], y[60000:]`

we get this result:
![bef](https://user-images.githubusercontent.com/66701689/142029066-d2c9a35f-3e6e-419f-90d7-993745d3e507.png)


and with this code

`import matplotlib.pyplot as plt
from sklearn.datasets import fetch_openml
from sklearn.exceptions import ConvergenceWarning
from sklearn.neural_network import MLPClassifier
from sklearn.model_selection import train_test_split

# Load data from https://www.openml.org/d/554
X, y = fetch_openml("mnist_784", version=1, return_X_y=True)
X = X / 255.0

# Split data into train partition and test partition
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0, test_size=0.7)`

we get this result:

![aft](https://user-images.githubusercontent.com/66701689/142029165-e8666144-5141-4e2b-ab25-c75fc28c1c29.png)

So, you were right. ``train_test_split`` is great. I tried going with less than 30 % train data but it doesn't get much faster, only the ``validation accuracy`` gets slightly worse (at least different from the original behavior)...

Commit:
Update plot_mnist_filters.py

adrinjalali reviewed (approved):
LGTM, you okay with the changes @ogrisel ?

adrinjalali commented:
Looking at the changes since @ogrisel 's approval, I think we can merge, thanks @sveneschlbeck
